model/riscv_platform.sail

```diff
@@ -426,7 +426,15 @@ function mmio_write forall 'n, 0 <'n <= max_mem_access . (paddr : physaddr, widt
 
 /* Platform initialization and ticking. */
 
+function plat_vendorid() -> bits(32) = to_bits(32, config platform.vendorid : int)
+function plat_archid() -> xlenbits = to_bits(xlen, config platform.archid : int)
+function plat_impid() -> xlenbits = to_bits(xlen, config platform.impid : int)
+
 function init_platform() -> unit = {
+  mvendorid = plat_vendorid();
```

Comment:
Maybe just use them directly?

```suggestion
  mvendorid = config platform.vendorid
  marchid = config platform.archid
  mimpid = config platform.impid
```

@Alasdair I might have mentioned this already but it would be nice if the config system had some way of zero-extending bit vectors (either from integers or from "0x123" strings. The downside of the above code is that you have to add/remove a load of extra zeros when you change XLEN for `archid` and `impid`.

Reply:
Right, wouldn't the advantage of the `plat_*()` calls be that they could use whatever `xlen` had been initialized to once we have #870?

Reply:
So directly assigning
```
mvendorid = config platform.vendorid;
```
results in
```
Pattern match failure in init_platform
```
while
```
mvendorid = plat_vendorid();
```
works.  I'll leave the `plat_*` functions in for now.
```

Reply:
You need to change the config to be `"0x00000000"` instead of `0`. Or leave it as 0 but then use `to_bits(config ...)`.

The config error messages are not especially clear yet!

Reply:
The second option works, thanks, that's nicer than what I had.  I couldn't get the first one to work.

Reply:
Actually you have to use

```
"impid": {
   "len": 64,
   "value": "0xF..."
 }
 ```
 
 See `writable_hpm_counters` for an example. This kind of sucks because you have to change all of the `len`s when you change XLEN, but I just discovered that after #870 I *think* you can do this:
 
```
 "impid": {
   "len": "xlen",
   "value": "34" // It can also be hex or binary and it gets zero extended (and truncated).
 }
```

I'll try it, but I think let's go with what you've got for now at least. Just having an integer is a bit nicer for the config file anyway.